File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/cot/Java/Python/s612878490_comments_removed.py
import sys

def main():
    line = input().strip()
    N, M = map(int, line.split())

    shortage = []
    for i in range(M):
        line = input().strip()
        A = int(line.split()[0])
        shortage.append(max(N - A, 0))

    shortage.sort()

    ans = sum(shortage[:-1])

    print(ans)

if __name__ == "__main__":
    main()  # End of Code

File: tasks/translation/output/Qwen/Qwen2.5-32B-Instruct-AWQ/codenet/cot/Java/C++/s612878490_comments_removed.cpp
#include <iostream>
#include <algorithm>
#include <vector>

int main() {
    std::string line;
    std::getline(std::cin, line);
    int N, M;
    std::sscanf(line.c_str(), "%d %d", &N, &M);

    std::vector<int> shortage(M);

    for (int i = 0; i < M; i++) {
        std::getline(std::cin, line);
        int A;
        std::sscanf(line.c_str(), "%d", &A);
        shortage[i] = N > A ? N - A : 0;
    }

    std::sort(shortage.begin(), shortage.end());

    int ans = 0;

    for (int i = 0; i < M - 1; i++) {
        ans += shortage[i];
    }

    std::cout << ans << std::endl;

    // End of Code
